firestore/query.go

```diff
@@ -364,6 +366,97 @@ func (q Query) Deserialize(bytes []byte) (Query, error) {
 	return q.fromProto(&runQueryRequest)
 }
 
+// DistanceMeasure is the distance measure to use when comparing vectors.
+type DistanceMeasure int32
+
+const (
+	// Measures the EUCLIDEAN distance between the vectors. See
+	// [Euclidean](https://en.wikipedia.org/wiki/Euclidean_distance) to learn
+	// more
+	DistanceMeasureEuclidean DistanceMeasure = 1
+
+	// Compares vectors based on the angle between them, which allows you to
+	// measure similarity that isn't based on the vectors magnitude.
+	// We recommend using DOT_PRODUCT with unit normalized vectors instead of
+	// COSINE distance, which is mathematically equivalent with better
+	// performance. See [Cosine
+	// Similarity](https://en.wikipedia.org/wiki/Cosine_similarity) to learn
+	// more.
+	DistanceMeasureCosine DistanceMeasure = 2
+
+	// Similar to cosine but is affected by the magnitude of the vectors. See
+	// [Dot Product](https://en.wikipedia.org/wiki/Dot_product) to learn more.
+	DistanceMeasureDotProduct DistanceMeasure = 3
+)
+
+// FindNearestOpts is options to use while building FindNearest vector query
+type FindNearestOpts struct {
```

Comment:
I'm not sure if this is the way to go. These aren't really options, they're both required, as far as I've been able to tell. I think they should be parameters to FindNearest.

Do you know of plans to add more options? If so, then something like this might make sense.

Reply:
Yes, there are plans to extend. E.g. https://github.com/googleapis/nodejs-firestore/issues/2079

Reply:
Modified as per the discussion